firedrake/mg/interface.py

```diff
@@ -18,20 +18,41 @@ def prolong(coarse, fine):
     hierarchy, lvl = utils.get_level(cfs)
     if hierarchy is None:
         raise RuntimeError("Coarse function not from hierarchy")
+    # for skipping hierarchies
+    if hasattr(hierarchy, '_full_hierarchy') == 1:
+        lvl = lvl * hierarchy._skip
+        hierarchy = hierarchy._full_hierarchy
```

Comment:
Cool. I'll make the transfer of the functionality to firedrake/mg. Should this be in a separate pull request or the same as this?